[BUG] gost lint fail

### ✅ 验证清单

- [x] 🔍 我已经搜索过 [现有 Issues](https://github.com/apache/dubbo-go/issues)，确信这不是重复问题

run golangci-lint in gost, lint fail

```
../../go/pkg/mod/github.com/dubbogo/gost@v1.14.3/database/kv/zk/client.go:200:33: cannot use 1 (untyped int constant) as *testing.T value in argument to zk.StartTestCluster
../../go/pkg/mod/github.com/dubbogo/gost@v1.14.3/database/kv/zk/client.go:200:36: cannot use nil as int value in argument to zk.StartTestCluster
../../go/pkg/mod/github.com/dubbogo/gost@v1.14.3/database/kv/zk/client.go:200:49: undefined: zk.WithRetryTimes)))
```

--- a/bytes/buffer.go
+++ b/bytes/buffer.go
@@ -331,7 +331,8 @@ func (b *Buffer) WriteByte(c byte) error {
 func (b *Buffer) WriteRune(r rune) (n int, err error) {
 	// Compare as uint32 to correctly handle negative runes.
 	if uint32(r) < utf8.RuneSelf {
-		b.WriteByte(byte(r))
+		// WriteByte returns nil for bytes.Buffer
+		_ = b.WriteByte(byte(r))
 		return 1, nil
 	}
 	b.lastRead = opInvalid

--- a/bytes/buffer.go
+++ b/bytes/buffer.go
@@ -219,7 +219,7 @@ func (b *Buffer) WriteNextEnd(n int) (int, error) {
 	bufLen := len(b.buf)
 	l := bufLen + n
 	if l > peekBufLen {
-		return 0, fmt.Errorf("U have not invoked @WriteNextBegin")
+		return 0, fmt.Errorf("you have not invoked @WriteNextBegin")
 	}
 
 	b.lastRead = opInvalid

--- a/bytes/buffer_test.go
+++ b/bytes/buffer_test.go
@@ -31,7 +31,8 @@ import (
 
 func TestBufferWithPeek(t *testing.T) {
 	var b Buffer
-	b.WriteString("hello")
+	_, err := b.WriteString("hello")
+	assert.Nil(t, err)
 
 	b1 := b
 	b1.WriteNextBegin(100)

--- a/bytes/bytes_pool.go
+++ b/bytes/bytes_pool.go
@@ -65,7 +65,7 @@ func (bp *BytesPool) findIndex(size int) int {
 func (bp *BytesPool) AcquireBytes(size int) *[]byte {
 	idx := bp.findIndex(size)
 	if idx >= bp.length {
-		buf := make([]byte, size, size)
+		buf := make([]byte, size)
 		return &buf
 	}
 

--- a/container/chan/unbounded_chan_test.go
+++ b/container/chan/unbounded_chan_test.go
@@ -37,7 +37,7 @@ func TestUnboundedChan(t *testing.T) {
 	}
 
 	for i := 1; i < 60; i++ {
-		v, _ := <-ch.Out()
+		v := <-ch.Out()
 		count += v.(int)
 	}
 
@@ -222,9 +222,7 @@ func BenchmarkUnboundedChan_Fixed(b *testing.B) {
 
 	b.RunParallel(func(pb *testing.PB) {
 		for pb.Next() {
-			select {
-			case ch.In() <- 1:
-			}
+			ch.In() <- 1
 
 			<-ch.Out()
 		}
@@ -238,9 +236,7 @@ func BenchmarkUnboundedChan_Extension(b *testing.B) {
 
 	b.RunParallel(func(pb *testing.PB) {
 		for pb.Next() {
-			select {
-			case ch.In() <- 1:
-			}
+			ch.In() <- 1
 
 			<-ch.Out()
 		}
@@ -254,9 +250,7 @@ func BenchmarkUnboundedChan_ExtensionUnlimited(b *testing.B) {
 
 	b.RunParallel(func(pb *testing.PB) {
 		for pb.Next() {
-			select {
-			case ch.In() <- 1:
-			}
+			ch.In() <- 1
 
 			<-ch.Out()
 		}

--- a/container/queue/queue_test.go
+++ b/container/queue/queue_test.go
@@ -31,7 +31,8 @@ import (
 func TestPut(t *testing.T) {
 	q := New(10)
 
-	q.Put(`test`)
+	err := q.Put(`test`)
+	assert.Nil(t, err)
 	assert.Equal(t, int64(1), q.Len())
 
 	results, err := q.Get(1)
@@ -41,7 +42,8 @@ func TestPut(t *testing.T) {
 	assert.Equal(t, `test`, result)
 	assert.True(t, q.Empty())
 
-	q.Put(`test2`)
+	err = q.Put(`test2`)
+	assert.Nil(t, err)
 	assert.Equal(t, int64(1), q.Len())
 
 	results, err = q.Get(1)
@@ -55,7 +57,8 @@ func TestPut(t *testing.T) {
 func TestGet(t *testing.T) {
 	q := New(10)
 
-	q.Put(`test`)
+	err := q.Put(`test`)
+	assert.Nil(t, err)
 	result, err := q.Get(2)
 	if !assert.Nil(t, err) {
 		return
@@ -65,8 +68,10 @@ func TestGet(t *testing.T) {
 	assert.Equal(t, `test`, result[0])
 	assert.Equal(t, int64(0), q.Len())
 
-	q.Put(`1`)
-	q.Put(`2`)
+	err = q.Put(`1`)
+	assert.Nil(t, err)
+	err = q.Put(`2`)
+	assert.Nil(t, err)
 
 	result, err = q.Get(1)
 	if !assert.Nil(t, err) {
@@ -89,9 +94,11 @@ func TestPoll(t *testing.T) {
 	q := New(10)
 
 	// should be able to Poll() before anything is present, without breaking future Puts
-	q.Poll(1, time.Millisecond)
+	// testing timeout behavior on empty queue
+	_, _ = q.Poll(1, time.Millisecond)
 
-	q.Put(`test`)
+	err := q.Put(`test`)
+	assert.Nil(t, err)
 	result, err := q.Poll(2, 0)
 	if !assert.Nil(t, err) {
 		return
@@ -101,8 +108,10 @@ func TestPoll(t *testing.T) {
 	assert.Equal(t, `test`, result[0])
 	assert.Equal(t, int64(0), q.Len())
 
-	q.Put(`1`)
-	q.Put(`2`)
+	err = q.Put(`1`)
+	assert.Nil(t, err)
+	err = q.Put(`2`)
+	assert.Nil(t, err)
 
 	result, err = q.Poll(1, time.Millisecond)
 	if !assert.Nil(t, err) {
@@ -135,7 +144,8 @@ func TestPollNoMemoryLeak(t *testing.T) {
 
 	for i := 0; i < 10; i++ {
 		// Poll() should cleanup waiters after timeout
-		q.Poll(1, time.Nanosecond)
+		// testing waiter cleanup mechanism
+		_, _ = q.Poll(1, time.Nanosecond)
 		assert.Len(t, q.waiters, 0)
 	}
 }
@@ -143,7 +153,8 @@ func TestPollNoMemoryLeak(t *testing.T) {
 func TestAddEmptyPut(t *testing.T) {
 	q := New(10)
 
-	q.Put()
+	// testing empty parameters behavior
+	_ = q.Put()
 
 	if q.Len() != 0 {
 		t.Errorf(`Expected len: %d, received: %d`, 0, q.Len())
@@ -153,7 +164,8 @@ func TestAddEmptyPut(t *testing.T) {
 func TestGetNonPositiveNumber(t *testing.T) {
 	q := New(10)
 
-	q.Put(`test`)
+	err := q.Put(`test`)
+	assert.Nil(t, err)
 	result, err := q.Get(0)
 	if !assert.Nil(t, err) {
 		return
@@ -171,7 +183,8 @@ func TestEmpty(t *testing.T) {
 		t.Errorf(`Expected empty queue.`)
 	}
 
-	q.Put(`test`)
+	err := q.Put(`test`)
+	assert.Nil(t, err)
 	if q.Empty() {
 		t.Errorf(`Expected non-empty queue.`)
 	}
@@ -182,7 +195,8 @@ func TestGetEmpty(t *testing.T) {
 
 	go func() {
 		time.Sleep(time.Second)
-		q.Put(`a`)
+		// test setup, not test target
+		_ = q.Put(`a`)
 	}()
 
 	result, err := q.Get(2)
@@ -219,7 +233,8 @@ func TestMultipleGetEmpty(t *testing.T) {
 	wg.Wait()
 	wg.Add(2)
 
-	q.Put(`a`, `b`, `c`)
+	err := q.Put(`a`, `b`, `c`)
+	assert.Nil(t, err)
 	wg.Wait()
 
 	if assert.Len(t, results[0], 1) && assert.Len(t, results[1], 1) {
@@ -238,7 +253,8 @@ func TestDispose(t *testing.T) {
 
 	// when the queue is not empty
 	q = New(10)
-	q.Put(`1`)
+	err := q.Put(`1`)
+	assert.Nil(t, err)
 	itemsDisposed = q.Dispose()
 
 	expected := []interface{}{`1`}
@@ -305,7 +321,8 @@ func BenchmarkQueue(b *testing.B) {
 
 	go func() {
 		for {
-			q.Get(1)
+			// benchmark focuses on throughput
+			_, _ = q.Get(1)
 			i++
 			if i == b.N {
 				wg.Done()
@@ -315,7 +332,8 @@ func BenchmarkQueue(b *testing.B) {
 	}()
 
 	for i := 0; i < b.N; i++ {
-		q.Put(`a`)
+		// benchmark focuses on throughput
+		_ = q.Put(`a`)
 	}
 
 	wg.Wait()
@@ -347,9 +365,12 @@ func BenchmarkChannel(b *testing.B) {
 
 func TestPeek(t *testing.T) {
 	q := New(10)
-	q.Put(`a`)
-	q.Put(`b`)
-	q.Put(`c`)
+	err := q.Put(`a`)
+	assert.Nil(t, err)
+	err = q.Put(`b`)
+	assert.Nil(t, err)
+	err = q.Put(`c`)
+	assert.Nil(t, err)
 	peekResult, err := q.Peek()
 	peekExpected := `a`
 	assert.Nil(t, err)
@@ -373,7 +394,8 @@ func TestPeekOnDisposedQueue(t *testing.T) {
 
 func TestGetUntil(t *testing.T) {
 	q := New(10)
-	q.Put(`a`, `b`, `c`)
+	err := q.Put(`a`, `b`, `c`)
+	assert.Nil(t, err)
 	result, err := q.GetUntil(func(item interface{}) bool {
 		return item != `c`
 	})
@@ -402,7 +424,8 @@ func TestGetUntilEmptyQueue(t *testing.T) {
 
 func TestGetUntilThenGet(t *testing.T) {
 	q := New(10)
-	q.Put(`a`, `b`, `c`)
+	err := q.Put(`a`, `b`, `c`)
+	assert.Nil(t, err)
 	takeItems, _ := q.GetUntil(func(item interface{}) bool {
 		return item != `c`
 	})
@@ -414,7 +437,8 @@ func TestGetUntilThenGet(t *testing.T) {
 
 func TestGetUntilNoMatches(t *testing.T) {
 	q := New(10)
-	q.Put(`a`, `b`, `c`)
+	err := q.Put(`a`, `b`, `c`)
+	assert.Nil(t, err)
 	takeItems, _ := q.GetUntil(func(item interface{}) bool {
 		return item != `a`
 	})
@@ -500,7 +524,8 @@ func TestWaiters(t *testing.T) {
 func TestExecuteInParallel(t *testing.T) {
 	q := New(10)
 	for i := 0; i < 10; i++ {
-		q.Put(i)
+		err := q.Put(i)
+		assert.Nil(t, err)
 	}
 
 	numCalls := uint64(0)
@@ -537,7 +562,8 @@ func BenchmarkQueuePut(b *testing.B) {
 	for i := 0; i < b.N; i++ {
 		q := qs[i]
 		for j := int64(0); j < numItems; j++ {
-			q.Put(j)
+			// benchmark focuses on throughput
+			_ = q.Put(j)
 		}
 	}
 }
@@ -550,7 +576,8 @@ func BenchmarkQueueGet(b *testing.B) {
 	for i := 0; i < b.N; i++ {
 		q := New(numItems)
 		for j := int64(0); j < numItems; j++ {
-			q.Put(j)
+			// benchmark setup
+			_ = q.Put(j)
 		}
 		qs = append(qs, q)
 	}
@@ -560,7 +587,8 @@ func BenchmarkQueueGet(b *testing.B) {
 	for i := 0; i < b.N; i++ {
 		q := qs[i]
 		for j := int64(0); j < numItems; j++ {
-			q.Get(1)
+			// benchmark focuses on throughput
+			_, _ = q.Get(1)
 		}
 	}
 }
@@ -573,7 +601,8 @@ func BenchmarkQueuePoll(b *testing.B) {
 	for i := 0; i < b.N; i++ {
 		q := New(numItems)
 		for j := int64(0); j < numItems; j++ {
-			q.Put(j)
+			// benchmark setup
+			_ = q.Put(j)
 		}
 		qs = append(qs, q)
 	}
@@ -582,7 +611,8 @@ func BenchmarkQueuePoll(b *testing.B) {
 
 	for _, q := range qs {
 		for j := int64(0); j < numItems; j++ {
-			q.Poll(1, time.Millisecond)
+			// benchmark focuses on throughput
+			_, _ = q.Poll(1, time.Millisecond)
 		}
 	}
 }
@@ -595,7 +625,8 @@ func BenchmarkExecuteInParallel(b *testing.B) {
 	for i := 0; i < b.N; i++ {
 		q := New(numItems)
 		for j := int64(0); j < numItems; j++ {
-			q.Put(j)
+			// benchmark setup
+			_ = q.Put(j)
 		}
 		qs = append(qs, q)
 	}

--- a/database/kv/etcd/v3/client.go
+++ b/database/kv/etcd/v3/client.go
@@ -175,7 +175,8 @@ func NewClientWithOptions(ctx context.Context, opts *Options) (*Client, error) {
 // NOTICE: need to get the lock before calling this method
 func (c *Client) clean() {
 	// close raw client
-	c.rawClient.Close()
+	// cleanup prioritizes resource release
+	_ = c.rawClient.Close()
 
 	// cancel ctx for raw client
 	c.cancel()
@@ -456,7 +457,8 @@ func (c *Client) keepAliveKV(k string, v string) error {
 
 	keepAlive, err := rawClient.KeepAlive(c.ctx, lease.ID)
 	if err != nil || keepAlive == nil {
-		rawClient.Revoke(c.ctx, lease.ID)
+		// prioritize returning keepalive failure
+		_, _ = rawClient.Revoke(c.ctx, lease.ID)
 		if err != nil {
 			return perrors.WithMessage(err, "keep alive lease")
 		}

--- a/database/kv/etcd/v3/client_test.go
+++ b/database/kv/etcd/v3/client_test.go
@@ -136,7 +136,9 @@ func (suite *ClientTestSuite) setUpClient() *Client {
 // set up a client for suite
 func (suite *ClientTestSuite) SetupTest() {
 	c := suite.setUpClient()
-	c.CleanKV()
+	if err := c.CleanKV(); err != nil {
+		suite.T().Fatal("CleanKV failed:", err)
+	}
 	suite.client = c
 	return
 }

--- a/database/kv/etcd/v3/client_test.go
+++ b/database/kv/etcd/v3/client_test.go
@@ -111,7 +111,6 @@ func (suite *ClientTestSuite) SetupSuite() {
 	}
 
 	suite.etcd = e
-	return
 }
 
 // stop etcd server
@@ -140,7 +139,6 @@ func (suite *ClientTestSuite) SetupTest() {
 		suite.T().Fatal("CleanKV failed:", err)
 	}
 	suite.client = c
-	return
 }
 
 func (suite *ClientTestSuite) TestClientClose() {

--- a/database/kv/zk/client.go
+++ b/database/kv/zk/client.go
@@ -214,53 +214,42 @@ func NewMockZookeeperClient(name string, timeout time.Duration, opts ...Option) 
 
 // HandleZkEvent handles zookeeper events
 func (d *DefaultHandler) HandleZkEvent(z *ZookeeperClient) {
-	var (
-		ok    bool
-		state int
-		event zk.Event
-	)
-	for {
-		select {
-		case event, ok = <-z.Session:
-			if !ok {
-				// channel already closed
-				return
-			}
-			switch event.State {
-			case zk.StateDisconnected:
-				atomic.StoreUint32(&z.valid, 0)
-			case zk.StateConnected:
-				z.eventRegistryLock.RLock()
-				for path, a := range z.eventRegistry {
-					if strings.HasPrefix(event.Path, path) {
-						for _, e := range a {
-							e <- event
-						}
-					}
-				}
-				z.eventRegistryLock.RUnlock()
-			case zk.StateConnecting, zk.StateHasSession:
-				if state == (int)(zk.StateHasSession) {
-					continue
-				}
-				if event.State == zk.StateHasSession {
-					atomic.StoreUint32(&z.valid, 1)
-					//if this is the first connection, don't trigger reconnect event
-					if !atomic.CompareAndSwapUint32(&z.initialized, 0, 1) {
-						close(z.reconnectCh)
-						z.reconnectCh = make(chan struct{})
-					}
-				}
-				z.eventRegistryLock.RLock()
-				if a, ok := z.eventRegistry[event.Path]; ok && 0 < len(a) {
+	var state int
+	for event := range z.Session {
+		switch event.State {
+		case zk.StateDisconnected:
+			atomic.StoreUint32(&z.valid, 0)
+		case zk.StateConnected:
+			z.eventRegistryLock.RLock()
+			for path, a := range z.eventRegistry {
+				if strings.HasPrefix(event.Path, path) {
 					for _, e := range a {
 						e <- event
 					}
 				}
-				z.eventRegistryLock.RUnlock()
 			}
-			state = (int)(event.State)
+			z.eventRegistryLock.RUnlock()
+		case zk.StateConnecting, zk.StateHasSession:
+			if state == (int)(zk.StateHasSession) {
+				continue
+			}
+			if event.State == zk.StateHasSession {
+				atomic.StoreUint32(&z.valid, 1)
+				//if this is the first connection, don't trigger reconnect event
+				if !atomic.CompareAndSwapUint32(&z.initialized, 0, 1) {
+					close(z.reconnectCh)
+					z.reconnectCh = make(chan struct{})
+				}
+			}
+			z.eventRegistryLock.RLock()
+			if a, ok := z.eventRegistry[event.Path]; ok && 0 < len(a) {
+				for _, e := range a {
+					e <- event
+				}
+			}
+			z.eventRegistryLock.RUnlock()
 		}
+		state = (int)(event.State)
 	}
 }
 
@@ -303,10 +292,7 @@ func (z *ZookeeperClient) UnregisterEvent(zkPath string, event chan zk.Event) {
 
 // ZkConnValid validates zookeeper connection
 func (z *ZookeeperClient) ZkConnValid() bool {
-	if atomic.LoadUint32(&z.valid) == 1 {
-		return true
-	}
-	return false
+	return atomic.LoadUint32(&z.valid) == 1
 }
 
 // Create will create the node recursively, which means that if the parent node is absent,

--- a/database/kv/zk/client_test.go
+++ b/database/kv/zk/client_test.go
@@ -78,7 +78,7 @@ func Test_getZookeeperClient(t *testing.T) {
 	client2.Close()
 	client3.Close()
 	client4.Close()
-	tc.Stop()
+	_ = tc.Stop()
 }
 
 func Test_Close(t *testing.T) {
@@ -126,7 +126,7 @@ func Test_Close(t *testing.T) {
 	client6.Close()
 	assert.Equal(t, client6.activeNumber, uint32(0))
 	assert.Equal(t, client6.Conn, (*zk.Conn)(nil))
-	tc.Stop()
+	_ = tc.Stop()
 }
 
 func Test_newMockZookeeperClient(t *testing.T) {

--- a/log/color.go
+++ b/log/color.go
@@ -70,18 +70,22 @@ func funcFileLine() string {
 func CPrintf(color []byte, format string, args ...interface{}) {
 	logStr := fmt.Sprintf(format, args...)
 	if isatty.IsTerminal(os.Stdout.Fd()) {
-		fmt.Fprintf(os.Stdout, string(color)+funcFileLine()+"%s"+string(reset), logStr)
+		// print error is non-critical
+		_, _ = fmt.Fprintf(os.Stdout, string(color)+funcFileLine()+"%s"+string(reset), logStr)
 	} else {
-		fmt.Fprintf(os.Stdout, "%s", logStr)
+		// print error is non-critical
+		_, _ = fmt.Fprintf(os.Stdout, "%s", logStr)
 	}
 }
 
 func CPrintfln(color []byte, format string, args ...interface{}) {
 	logStr := fmt.Sprintf(format, args...)
 	if isatty.IsTerminal(os.Stdout.Fd()) {
-		fmt.Fprintf(os.Stdout, string(color)+funcFileLine()+"%s"+string(reset)+"\n", logStr)
+		// print error is non-critical
+		_, _ = fmt.Fprintf(os.Stdout, string(color)+funcFileLine()+"%s"+string(reset)+"\n", logStr)
 	} else {
-		fmt.Fprintf(os.Stdout, "%s\n", logStr)
+		// print error is non-critical
+		_, _ = fmt.Fprintf(os.Stdout, "%s\n", logStr)
 	}
 }
 

--- a/log/logger/logger.go
+++ b/log/logger/logger.go
@@ -164,9 +164,10 @@ func initZapLoggerWithSyncer(conf *Config) *zap.Logger {
 
 // getEncoder get encoder by config, zapcore support json and console encoder
 func (c *Config) getEncoder() zapcore.Encoder {
-	if c.ZapConfig.Encoding == "json" {
+	switch c.ZapConfig.Encoding {
+	case "json":
 		return zapcore.NewJSONEncoder(c.ZapConfig.EncoderConfig)
-	} else if c.ZapConfig.Encoding == "console" {
+	case "console":
 		return zapcore.NewConsoleEncoder(c.ZapConfig.EncoderConfig)
 	}
 	return nil

--- a/log/pretty.go
+++ b/log/pretty.go
@@ -42,13 +42,16 @@ func ColorSprintf(fmt string, args ...interface{}) string {
 }
 
 func ColorPrint(i interface{}) {
-	pp.Print(i)
+	// print error is non-critical
+	_, _ = pp.Print(i)
 }
 
 func ColorPrintln(i interface{}) {
-	pp.Println(i)
+	// print error is non-critical
+	_, _ = pp.Println(i)
 }
 
 func ColorPrintf(fmt string, args ...interface{}) {
-	pp.Printf(fmt, args...)
+	// print error is non-critical
+	_, _ = pp.Printf(fmt, args...)
 }

--- a/math/big/decimal.go
+++ b/math/big/decimal.go
@@ -979,10 +979,7 @@ func (d *Decimal) Round(to *Decimal, frac int, roundMode RoundMode) (err error) 
 			}
 		}
 	} else {
-		for {
-			if to.wordBuf[toIdx] != 0 {
-				break
-			}
+		for to.wordBuf[toIdx] == 0 {
 			if toIdx == 0 {
 				/* making 'zero' with the proper scale */
 				idx := wordsFracTo + 1
@@ -2078,10 +2075,7 @@ func DecimalMul(from1, from2, to *Decimal) error {
 	if to.negative {
 		idx := 0
 		end := wordsIntTo + wordsFracTo
-		for {
-			if to.wordBuf[idx] != 0 {
-				break
-			}
+		for to.wordBuf[idx] == 0 {
 			idx++
 			/* We got decimal zero */
 			if idx == end {

--- a/math/big/helper.go
+++ b/math/big/helper.go
@@ -146,10 +146,11 @@ func strToInt(str string) (int64, error) {
 	}
 	negative := false
 	i := 0
-	if str[i] == '-' {
+	switch str[i] {
+	case '-':
 		negative = true
 		i++
-	} else if str[i] == '+' {
+	case '+':
 		i++
 	}
 

--- a/math/big/integer_test.go
+++ b/math/big/integer_test.go
@@ -103,7 +103,7 @@ func TestInteger_FromSignAndMag(t *testing.T) {
 			}
 
 			sign, mag := s.GetSignAndMag()
-			if !(sign == tt.args.signum && reflect.DeepEqual(mag, tt.args.mag)) {
+			if sign != tt.args.signum || !reflect.DeepEqual(mag, tt.args.mag) {
 				t.Error("want ", tt.args.signum, tt.args.mag,
 					"got", sign, mag)
 			}

--- a/math/big/integer_test.go
+++ b/math/big/integer_test.go
@@ -113,7 +113,10 @@ func TestInteger_FromSignAndMag(t *testing.T) {
 
 func TestInteger_Json(t *testing.T) {
 	i := new(Integer)
-	i.FromString(`1234567`)
+	err := i.FromString(`1234567`)
+	if err != nil {
+		t.Fatal("FromString failed:", err)
+	}
 	bytes, err := json.Marshal(i.String())
 	if err != nil || string(bytes) != `"1234567"` {
 		t.Error(string(bytes), err)

--- a/net/net.go
+++ b/net/net.go
@@ -200,7 +200,7 @@ func matchIPRange(pattern, host, port string) bool {
 	}
 
 	pattern = strings.TrimSpace(pattern)
-	if "*.*.*.*" == pattern || "*" == pattern {
+	if pattern == "*.*.*.*" || pattern == "*" {
 		return true
 	}
 
@@ -240,7 +240,7 @@ func matchIPRange(pattern, host, port string) bool {
 
 	ipAddress := strings.Split(host, splitCharacter)
 	for i := 0; i < len(mask); i++ {
-		if "*" == mask[i] || mask[i] == ipAddress[i] {
+		if mask[i] == "*" || mask[i] == ipAddress[i] {
 			continue
 		} else if strings.Contains(mask[i], "-") {
 			rangeNumStrs := strings.Split(mask[i], "-")
@@ -254,7 +254,7 @@ func matchIPRange(pattern, host, port string) bool {
 			if ip < min || ip > max {
 				return false
 			}
-		} else if "0" == ipAddress[i] && "0" == mask[i] || "00" == mask[i] || "000" == mask[i] || "0000" == mask[i] {
+		} else if ipAddress[i] == "0" && mask[i] == "0" || mask[i] == "00" || mask[i] == "000" || mask[i] == "0000" {
 			continue
 		} else if mask[i] != ipAddress[i] {
 			return false

--- a/net/net_test.go
+++ b/net/net_test.go
@@ -60,7 +60,7 @@ func TestListenOnTCPRandomPort(t *testing.T) {
 	l, err := ListenOnTCPRandomPort("")
 	assert.Nil(t, err)
 	t.Logf("a tcp server listen on a random addr:%s", l.Addr())
-	l.Close()
+	_ = l.Close()
 
 	localIP, err := GetLocalIP()
 	if err == nil {
@@ -68,7 +68,7 @@ func TestListenOnTCPRandomPort(t *testing.T) {
 		assert.Nil(t, err)
 		assert.True(t, strings.Contains(l.Addr().String(), localIP))
 		t.Logf("a tcp server listen on a random addr:%s", l.Addr())
-		l.Close()
+		_ = l.Close()
 	}
 }
 
@@ -76,7 +76,7 @@ func TestListenOnUDPRandomPort(t *testing.T) {
 	l, err := ListenOnUDPRandomPort("")
 	assert.Nil(t, err)
 	t.Logf("a udp peer listen on a random addr:%s", l.LocalAddr())
-	l.Close()
+	_ = l.Close()
 
 	localIP, err := GetLocalIP()
 	if err == nil {
@@ -84,7 +84,7 @@ func TestListenOnUDPRandomPort(t *testing.T) {
 		assert.Nil(t, err)
 		assert.True(t, strings.Contains(l.LocalAddr().String(), localIP))
 		t.Logf("a udp server listen on a random addr:%s", l.LocalAddr())
-		l.Close()
+		_ = l.Close()
 	}
 }
 

--- a/runtime/sys.go
+++ b/runtime/sys.go
@@ -71,11 +71,7 @@ func GetMemoryStat() (total, used, free uint64, usedPercent float64) {
 // IsCgroup checks whether current os is a container or not
 func IsCgroup() bool {
 	ok, _ := gxfilepath.Exists(cgroupMemLimitPath)
-	if ok {
-		return true
-	}
-
-	return false
+	return ok
 }
 
 // GetCgroupMemoryLimit returns a container's total memory in bytes

--- a/runtime/sys.go
+++ b/runtime/sys.go
@@ -201,7 +201,10 @@ func readLinesFromFile(filepath string) []string {
 	if err != nil {
 		return res
 	}
-	defer f.Close()
+	defer func() {
+		// file is read-only
+		_ = f.Close()
+	}()
 	buff := bufio.NewReader(f)
 	for {
 		line, _, err := buff.ReadLine()

--- a/strings/slice.go
+++ b/strings/slice.go
@@ -21,6 +21,21 @@ import (
 	"unsafe"
 )
 
+// Slice converts a string to a byte slice without memory allocation.
+// It uses unsafe operations to directly reference the underlying string data,
+// avoiding the overhead of copying bytes.
+//
+// WARNING: This function uses unsafe operations. The returned byte slice
+// shares the same underlying memory as the string. Modifying the returned
+// slice will cause undefined behavior since strings are immutable in Go.
+//
+// Parameters:
+//   - s: The string to convert
+//
+// Returns a byte slice that references the string's underlying data.
+//
+// Use this function only when you need read-only access to string bytes
+// with zero allocation, and you understand the safety implications.
 func Slice(s string) []byte {
 	ptr := unsafe.StringData(s)
 	return unsafe.Slice((*byte)(ptr), len(s))

--- a/strings/strings.go
+++ b/strings/strings.go
@@ -23,6 +23,10 @@ import (
 	"strings"
 )
 
+// IsNil checks whether the given interface value is nil.
+// It performs a deep nil check by examining both the interface itself
+// and the underlying value using reflection.
+// Returns true if the value is nil or if the underlying value is nil.
 func IsNil(i interface{}) bool {
 	if i == nil {
 		return true
@@ -35,6 +39,19 @@ func IsNil(i interface{}) bool {
 	return false
 }
 
+// RegSplit splits a string by a regular expression pattern.
+// It uses the provided regex pattern to find all split positions in the text
+// and returns a slice of substrings between those positions.
+//
+// Parameters:
+//   - text: The string to be split
+//   - regexSplit: The regular expression pattern used as delimiter
+//
+// Returns a slice of strings containing the split results.
+//
+// Example:
+//
+//	RegSplit("a;b;c", "\\s*[;]+\\s*") returns ["a", "b", "c"]
 func RegSplit(text string, regexSplit string) []string {
 	reg := regexp.MustCompile(regexSplit)
 	indexes := reg.FindAllStringIndex(text, -1)

--- a/strings/strings.go
+++ b/strings/strings.go
@@ -27,16 +27,19 @@ import (
 // It performs a deep nil check by examining both the interface itself
 // and the underlying value using reflection.
 // Returns true if the value is nil or if the underlying value is nil.
+// For non-nilable types (e.g., int, string, struct), it returns false.
 func IsNil(i interface{}) bool {
 	if i == nil {
 		return true
 	}
 
-	if reflect.ValueOf(i).IsNil() {
-		return true
+	v := reflect.ValueOf(i)
+	switch v.Kind() {
+	case reflect.Chan, reflect.Func, reflect.Interface, reflect.Map, reflect.Pointer, reflect.Slice, reflect.UnsafePointer:
+		return v.IsNil()
+	default:
+		return false
 	}
-
-	return false
 }
 
 // RegSplit splits a string by a regular expression pattern.

--- a/strings/strings.go
+++ b/strings/strings.go
@@ -51,7 +51,7 @@ func RegSplit(text string, regexSplit string) []string {
 // IsMatchPattern is used to determine whether pattern
 // and value match with wildcards currently supported *
 func IsMatchPattern(pattern string, value string) bool {
-	if "*" == pattern {
+	if pattern == "*" {
 		return true
 	}
 	if len(pattern) == 0 && len(value) == 0 {

--- a/strings/strings_test.go
+++ b/strings/strings_test.go
@@ -44,3 +44,55 @@ func TestIsMatchPattern(t *testing.T) {
 	assert.Equal(t, true, IsMatchPattern("v*e", "value"))
 	assert.Equal(t, true, IsMatchPattern("val*e", "value"))
 }
+
+func TestIsNil(t *testing.T) {
+	// Test nil interface
+	assert.True(t, IsNil(nil))
+
+	// Test nilable types that are nil
+	var nilChan chan int
+	assert.True(t, IsNil(nilChan))
+
+	var nilFunc func()
+	assert.True(t, IsNil(nilFunc))
+
+	var nilInterface interface{}
+	assert.True(t, IsNil(nilInterface))
+
+	var nilMap map[string]int
+	assert.True(t, IsNil(nilMap))
+
+	var nilPtr *int
+	assert.True(t, IsNil(nilPtr))
+
+	var nilSlice []int
+	assert.True(t, IsNil(nilSlice))
+
+	// Test nilable types that are not nil
+	ch := make(chan int)
+	assert.False(t, IsNil(ch))
+
+	fn := func() {}
+	assert.False(t, IsNil(fn))
+
+	m := map[string]int{}
+	assert.False(t, IsNil(m))
+
+	i := 42
+	ptr := &i
+	assert.False(t, IsNil(ptr))
+
+	s := []int{}
+	assert.False(t, IsNil(s))
+
+	// Test non-nilable types (should not panic)
+	assert.False(t, IsNil(42))
+	assert.False(t, IsNil("hello"))
+	assert.False(t, IsNil(true))
+	assert.False(t, IsNil(3.14))
+
+	type testStruct struct {
+		field int
+	}
+	assert.False(t, IsNil(testStruct{field: 1}))
+}

--- a/sync/base_worker_pool.go
+++ b/sync/base_worker_pool.go
@@ -171,26 +171,20 @@ func (p *baseWorkerPool) worker(workerId int, wg *sync.WaitGroup) {
 	chanId := workerId % len(p.taskQueues)
 
 	wg.Done()
-	for {
-		select {
-		case t, ok := <-p.taskQueues[chanId]:
-			if !ok {
-				return
-			}
-			if t != nil {
-				func() {
-					// prevent from goroutine panic
-					defer func() {
-						if r := recover(); r != nil {
-							if p.logger != nil {
-								p.logger.Errorf("goroutine panic: %v\n%s", r, string(debug.Stack()))
-							}
+	for t := range p.taskQueues[chanId] {
+		if t != nil {
+			func() {
+				// prevent from goroutine panic
+				defer func() {
+					if r := recover(); r != nil {
+						if p.logger != nil {
+							p.logger.Errorf("goroutine panic: %v\n%s", r, string(debug.Stack()))
 						}
-					}()
-					// execute task
-					t()
+					}
 				}()
-			}
+				// execute task
+				t()
+			}()
 		}
 	}
 }

--- a/time/sleep_test.go
+++ b/time/sleep_test.go
@@ -47,14 +47,9 @@ func TestNewTimerWheel(t *testing.T) {
 	}()
 
 	cw.Start()
-	for {
-		select {
-		case <-wheel.After(TimeMillisecondDuration(100)):
-			index++
-			if index >= 10 {
-				return
-			}
-		}
+	for index < 10 {
+		<-wheel.After(TimeMillisecondDuration(100))
+		index++
 	}
 }
 
@@ -82,14 +77,9 @@ func TestAfter(t *testing.T) {
 		}()
 
 		cw.Start()
-		for {
-			select {
-			case <-wheel.After(d):
-				index++
-				if index >= num {
-					return
-				}
-			}
+		for index < num {
+			<-wheel.After(d)
+			index++
 		}
 	}
 

--- a/time/timer.go
+++ b/time/timer.go
@@ -580,14 +580,13 @@ func (w *TimerWheel) NewTimer(d time.Duration) *Timer {
 
 // After waits for the duration to elapse and then sends the current time
 // on the returned channel.
+// Returns nil if the timer cannot be created (e.g., timer queue is full).
 func (w *TimerWheel) After(d time.Duration) <-chan time.Time {
-	//timer := defaultTimer.NewTimer(d)
-	//if timer == nil {
-	//	return nil
-	//}
-	//
-	//return timer.C
-	return w.NewTimer(d).C
+	t := w.NewTimer(d)
+	if t == nil {
+		return nil
+	}
+	return t.C
 }
 
 func goFunc(_ TimerID, _ time.Time, arg interface{}) error {
@@ -599,16 +598,21 @@ func goFunc(_ TimerID, _ time.Time, arg interface{}) error {
 // AfterFunc waits for the duration to elapse and then calls f
 // in its own goroutine. It returns a Timer that can
 // be used to cancel the call using its Stop method.
+// Returns nil if the timer cannot be created (e.g., timer queue is full).
 func (w *TimerWheel) AfterFunc(d time.Duration, f func()) *Timer {
 	t, _ := w.AddTimer(goFunc, TimerOnce, d, f)
-
 	return t
 }
 
 // Sleep pauses the current goroutine for at least the duration d.
 // A negative or zero duration causes Sleep to return immediately.
+// If the timer cannot be created, Sleep returns immediately.
 func (w *TimerWheel) Sleep(d time.Duration) {
-	<-w.NewTimer(d).C
+	t := w.NewTimer(d)
+	if t == nil {
+		return
+	}
+	<-t.C
 }
 
 ////////////////////////////////////////////////
@@ -650,7 +654,11 @@ func (w *TimerWheel) TickFunc(d time.Duration, f func()) *Ticker {
 // channel only. While Tick is useful for clients that have no need to shut down
 // the Ticker, be aware that without a way to shut it down the underlying
 // Ticker cannot be recovered by the garbage collector; it "leaks".
-// Unlike NewTicker, Tick will return nil if d <= 0.
+// Unlike NewTicker, Tick will return nil if d <= 0 or if the ticker cannot be created.
 func (w *TimerWheel) Tick(d time.Duration) <-chan time.Time {
-	return w.NewTicker(d).C
+	ticker := w.NewTicker(d)
+	if ticker == nil {
+		return nil
+	}
+	return ticker.C
 }

--- a/time/timer.go
+++ b/time/timer.go
@@ -464,6 +464,22 @@ func (w *TimerWheel) timerUpdate(curTime time.Time) int {
 	return 0
 }
 
+// enqueueTimerAction attempts to enqueue a timer action in a non-blocking way.
+// It returns an error if the operation would block, preventing potential deadlocks
+// when called from the timer wheel goroutine.
+func (w *TimerWheel) enqueueTimerAction(action *timerNodeAction) error {
+	if !w.enable.Load() {
+		return ErrTimeChannelClosed
+	}
+
+	select {
+	case w.timerQ.In() <- action:
+		return nil
+	default:
+		return errors.New("timer queue is full, cannot enqueue action without blocking")
+	}
+}
+
 // Stop stops the ticker
 func (w *TimerWheel) Stop() {
 	w.once.Do(func() {
@@ -509,7 +525,9 @@ func (w *TimerWheel) AddTimer(f TimerFunc, typ TimerType, period time.Duration, 
 
 	t := &Timer{w: w}
 	node := newTimerNode(f, typ, int64(period), arg)
-	w.timerQ.In() <- &timerNodeAction{node: node, action: TimerActionAdd}
+	if err := w.enqueueTimerAction(&timerNodeAction{node: node, action: TimerActionAdd}); err != nil {
+		return nil, err
+	}
 	t.ID = node.ID
 	return t, nil
 }
@@ -519,8 +537,7 @@ func (w *TimerWheel) deleteTimer(t *Timer) error {
 		return ErrTimeChannelClosed
 	}
 
-	w.timerQ.In() <- &timerNodeAction{action: TimerActionDel, node: &timerNode{ID: t.ID}}
-	return nil
+	return w.enqueueTimerAction(&timerNodeAction{action: TimerActionDel, node: &timerNode{ID: t.ID}})
 }
 
 func (w *TimerWheel) resetTimer(t *Timer, d time.Duration) error {
@@ -528,8 +545,7 @@ func (w *TimerWheel) resetTimer(t *Timer, d time.Duration) error {
 		return ErrTimeChannelClosed
 	}
 
-	w.timerQ.In() <- &timerNodeAction{action: TimerActionReset, node: &timerNode{ID: t.ID, period: int64(d)}}
-	return nil
+	return w.enqueueTimerAction(&timerNodeAction{action: TimerActionReset, node: &timerNode{ID: t.ID, period: int64(d)}})
 }
 
 func sendTime(_ TimerID, t time.Time, arg interface{}) error {

--- a/time/timer.go
+++ b/time/timer.go
@@ -220,14 +220,14 @@ func NewTimerWheel() *TimerWheel {
 
 				nodeAction := node.(*timerNodeAction)
 				// just one w.timerQ channel to ensure the exec sequence of timer event.
-				switch {
-				case nodeAction.action == TimerActionAdd:
+				switch nodeAction.action {
+				case TimerActionAdd:
 					w.number.Add(1)
 					w.insertTimerNode(nodeAction.node)
-				case nodeAction.action == TimerActionDel:
+				case TimerActionDel:
 					w.number.Add(-1)
 					w.deleteTimerNode(nodeAction.node)
-				case nodeAction.action == TimerActionReset:
+				case TimerActionReset:
 					// log.CInfo("node action:%#v", nodeAction)
 					w.resetTimerNode(nodeAction.node)
 				default:
@@ -449,7 +449,7 @@ func (w *TimerWheel) timerUpdate(curTime time.Time) int {
 
 	maxIdx = 0
 	for idx = 0; idx < maxTimerLevel; idx++ {
-		if 0 != inc[idx] {
+		if inc[idx] != 0 {
 			w.hand[idx] += inc[idx]
 			inc[idx+1] += w.hand[idx] / limit[idx]
 			w.hand[idx] %= limit[idx]
@@ -509,11 +509,9 @@ func (w *TimerWheel) AddTimer(f TimerFunc, typ TimerType, period time.Duration, 
 
 	t := &Timer{w: w}
 	node := newTimerNode(f, typ, int64(period), arg)
-	select {
-	case w.timerQ.In() <- &timerNodeAction{node: node, action: TimerActionAdd}:
-		t.ID = node.ID
-		return t, nil
-	}
+	w.timerQ.In() <- &timerNodeAction{node: node, action: TimerActionAdd}
+	t.ID = node.ID
+	return t, nil
 }
 
 func (w *TimerWheel) deleteTimer(t *Timer) error {
@@ -521,10 +519,8 @@ func (w *TimerWheel) deleteTimer(t *Timer) error {
 		return ErrTimeChannelClosed
 	}
 
-	select {
-	case w.timerQ.In() <- &timerNodeAction{action: TimerActionDel, node: &timerNode{ID: t.ID}}:
-		return nil
-	}
+	w.timerQ.In() <- &timerNodeAction{action: TimerActionDel, node: &timerNode{ID: t.ID}}
+	return nil
 }
 
 func (w *TimerWheel) resetTimer(t *Timer, d time.Duration) error {
@@ -532,10 +528,8 @@ func (w *TimerWheel) resetTimer(t *Timer, d time.Duration) error {
 		return ErrTimeChannelClosed
 	}
 
-	select {
-	case w.timerQ.In() <- &timerNodeAction{action: TimerActionReset, node: &timerNode{ID: t.ID, period: int64(d)}}:
-		return nil
-	}
+	w.timerQ.In() <- &timerNodeAction{action: TimerActionReset, node: &timerNode{ID: t.ID, period: int64(d)}}
+	return nil
 }
 
 func sendTime(_ TimerID, t time.Time, arg interface{}) error {

